testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveBucketedTables.java

```diff
@@ -289,16 +299,98 @@ public void testBucketingVersion()
         List<String> bucketV1NameOptions = ImmutableList.of(bucketV1);
         List<String> bucketV2NameOptions = ImmutableList.of(bucketV2Standard, bucketV2DirectInsert);
 
-        testBucketingVersion(BucketingType.BUCKETED_DEFAULT, value, false, (getHiveVersionMajor() < 3) ? bucketV1NameOptions : bucketV2NameOptions);
-        testBucketingVersion(BucketingType.BUCKETED_DEFAULT, value, true, (getHiveVersionMajor() < 3) ? bucketV1NameOptions : bucketV2NameOptions);
-        testBucketingVersion(BucketingType.BUCKETED_V1, value, false, bucketV1NameOptions);
-        testBucketingVersion(BucketingType.BUCKETED_V1, value, true, bucketV1NameOptions);
+        testBucketingVersion(BUCKETED_DEFAULT, value, false, (getHiveVersionMajor() < 3) ? bucketV1NameOptions : bucketV2NameOptions);
+        testBucketingVersion(BUCKETED_DEFAULT, value, true, (getHiveVersionMajor() < 3) ? bucketV1NameOptions : bucketV2NameOptions);
+        testBucketingVersion(BUCKETED_V1, value, false, bucketV1NameOptions);
+        testBucketingVersion(BUCKETED_V1, value, true, bucketV1NameOptions);
         if (getHiveVersionMajor() >= 3) {
             testBucketingVersion(BucketingType.BUCKETED_V2, value, false, bucketV2NameOptions);
             testBucketingVersion(BucketingType.BUCKETED_V2, value, true, bucketV2NameOptions);
         }
     }
 
+    @Test(dataProvider = "testBucketingWithUnsupportedDataTypesDataProvider")
+    public void testBucketingWithUnsupportedDataTypes(BucketingType bucketingType, String columnToBeBucketed)
+    {
+        try (TemporaryHiveTable table = temporaryHiveTable("table_with_unsupported_bucketing_types_" + randomTableSuffix())) {
+            String tableName = table.getName();
+            onHive().executeQuery(format("CREATE TABLE %s (" +
+                            "n_integer       INT," +
+                            "n_decimal       DECIMAL(9, 2)," +
+                            "n_timestamp     TIMESTAMP," +
+                            "n_char          CHAR(10)," +
+                            "n_binary        BINARY," +
+                            "n_union         UNIONTYPE<INT,STRING>," +
+                            "n_struct        STRUCT<field1:INT,field2:STRING>) " +
+                            "CLUSTERED BY (%s) INTO 2 BUCKETS " +
+                            "STORED AS ORC " +
+                            "%s",
+                    tableName,
+                    columnToBeBucketed,
+                    hiveTableProperties(bucketingType)));
+
+            populateRowToHiveTable(
+                    tableName,
+                    ImmutableList.<String>builder()
+                            .add("1")
+                            .add("CAST(1 AS DECIMAL(9, 2))")
+                            .add("current_timestamp()")
+                            .add("'Trino'")
```

Comment:
make it more useful when debugging the test:

```suggestion
                            .add("'some char'")
```